Extended param file can not be built when base contains params using variable substitution

**Bicep version**
0.38.33 (6bb5d5f859)

**Describe the bug**
I have a base bicepparam file which contains params that use variable substitution. When building a param file extending from this base I get a vague exception. The error is also reported as a fatal error by the vscode linter.

**To Reproduce**
Using below bicepparam files:
```bicep
// base.bicepparam
using none

param someParam1 = 'tst'
param someParam2 = 'my-value-${someParam1}'

// main.bicepparam
using 'main.bicep'

extends 'base.bicepparam'

// main.bicep
// empty
```

Doing `bicep build-params main.bicepparam` throws the following error:

```
Unhandled exception. System.NotImplementedException: Encountered an unexpected symbol kind '' and type '' when generating a variable access expression.
   at Bicep.Core.Intermediate.ExpressionBuilder.ConvertVariableAccess(VariableAccessSyntax variableAccessSyntax)
   at Bicep.Core.Intermediate.ExpressionBuilder.ConvertWithoutLowering(SyntaxBase syntax)
   at System.Linq.Enumerable.SelectArrayIterator`2.MoveNext()
   at System.Collections.Generic.List`1.AddRange(IEnumerable`1)
   at Bicep.Core.Intermediate.ExpressionBuilder.ConvertWithoutLowering(SyntaxBase syntax)
   at Bicep.Core.Intermediate.ExpressionBuilder.Convert(SyntaxBase syntax)
   at Bicep.Core.Emit.ExpressionConverter.ConvertToIntermediateExpression(SyntaxBase syntax)
   at Bicep.Core.Emit.ParameterAssignmentEvaluator.<EvaluateParameter>b__18_0(ParameterAssignmentSymbol parameter)
   at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd(TKey, Func`2)
   at Bicep.Core.Emit.ParameterAssignmentEvaluator.EvaluateParameter(ParameterAssignmentSymbol parameter)
   at Bicep.Core.Emit.EmitLimitationCalculator.CalculateParameterAssignments(SemanticModel model, ParameterAssignmentEvaluator evaluator, IDiagnosticWriter diagnostics)
   at Bicep.Core.Emit.EmitLimitationCalculator.Calculate(SemanticModel model)
   at Bicep.Core.Semantics.SemanticModel.<.ctor>b__20_2()
   at System.Lazy`1.ViaFactory(LazyThreadSafetyMode)
   at System.Lazy`1.ExecutionAndPublication(LazyHelper, Boolean)
   at System.Lazy`1.CreateValue()
   at System.Lazy`1.get_Value()
   at Bicep.Core.Semantics.SemanticModel.get_EmitLimitationInfo()
   at Bicep.Core.Semantics.SemanticModel.GetSemanticDiagnostics()
   at Bicep.Core.Semantics.SemanticModel.AssembleDiagnostics()
   at System.Lazy`1.ViaFactory(LazyThreadSafetyMode)
   at System.Lazy`1.ExecutionAndPublication(LazyHelper, Boolean)
   at System.Lazy`1.CreateValue()
   at System.Lazy`1.get_Value()
   at Bicep.Core.Semantics.SemanticModel.GetAllDiagnostics()
   at Bicep.Core.Semantics.Compilation.GetSourceFileDiagnostics(IOUri sourceFileUri)
   at Bicep.Core.Semantics.Compilation.GetSourceFileDiagnostics(ISourceFile sourceFile)
   at System.Collections.Immutable.ImmutableDictionary.<>c__DisplayClass9_0`3.<ToImmutableDictionary>b__0(TSource element)
   at System.Linq.Enumerable.SelectEnumerableIterator`2.MoveNext()
   at System.Collections.Immutable.ImmutableDictionary`2.AddRange(IEnumerable`1, MutationInput, KeyCollisionBehavior )
   at System.Collections.Immutable.ImmutableDictionary`2.AddRange(IEnumerable`1, Boolean)
   at System.Collections.Immutable.ImmutableDictionary`2.AddRange(IEnumerable`1 )
   at System.Collections.Immutable.ImmutableDictionary.ToImmutableDictionary[TSource,TKey,TValue](IEnumerable`1, Func`2, Func`2, IEqualityComparer`1 , IEqualityComparer`1 )
   at System.Collections.Immutable.ImmutableDictionary.ToImmutableDictionary[TSource,TKey,TValue](IEnumerable`1, Func`2, Func`2)       
   at Bicep.Core.Semantics.Compilation.GetAllDiagnosticsByBicepFile()
   at Bicep.Cli.Logging.DiagnosticLogger.LogDiagnostics(DiagnosticOptions options, Compilation compilation)
   at Bicep.Cli.Commands.BuildParamsCommand.Compile(ActiveSourceFileSet workspace, IOUri inputUri, IOUri bicepFileUri, IOUri outputUri, Boolean noRestore, Nullable`1 diagnosticsFormat, Boolean outputToStdOut)
   at Bicep.Cli.Commands.BuildParamsCommand.RunAsync(BuildParamsArguments args)
   at Bicep.Cli.Program.RunAsync(String[] args, CancellationToken cancellationToken)
   at Bicep.Cli.Program.<>c__DisplayClass3_0.<<Main>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at Bicep.Cli.Program.RunWithCancellationAsync(Func`2 runFunc)
   at Bicep.Cli.Program.Main(String[] args)
   at Bicep.Cli.Program.<Main>(String[] args)
```

**Additional context**
Removing the someParam2 parameter fixes the issue.

@polatengin I'm still getting the same error in my Azure DevOps pipelines, even when i have forced it to use the latest bicep version. Deploying the code locally works though.
```
- stage: Preview
  jobs:
  - job: Preview
    steps:
    - task: AzureCLI@2
      inputs:
        azureSubscription: $(serviceConnectionName)
        scriptType: 'bash'
        scriptLocation: 'inlineScript'
        inlineScript: |
          az bicep uninstall
          az config set bicep.use_binary_from_path=false
          az bicep install
          az bicep version
          az deployment sub what-if \
            --location $(deploymentDefaultLocation) \
            --template-file bicep/subscription/subscription.bicep \
            --parameters bicep/subscription/subscription.dev.bicepparam
```
Log:
```
Task         : Azure CLI
Description  : Run Azure CLI commands against an Azure subscription in a PowerShell Core/Shell script when running on Linux agent or PowerShell/PowerShell Core/Batch script when running on Windows agent.
Version      : 2.262.1
Author       : Microsoft Corporation
Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/deploy/azure-cli
==============================================================================
/usr/bin/az version
{
  "azure-cli": "2.79.0",
  "azure-cli-core": "2.79.0",
  "azure-cli-telemetry": "1.1.0",
  "extensions": {
    "azure-devops": "1.0.2"
  }
}
Setting AZURE_CONFIG_DIR env variable to: /home/vsts/work/_temp/.azclitask
Setting active cloud to: AzureCloud
/usr/bin/az cloud set -n AzureCloud
/usr/bin/az login --service-principal -u *** --tenant xxx --allow-no-subscriptions --federated-token ***
[
  {
    "cloudName": "AzureCloud",
    "homeTenantId": "xxx",
    "id": "xxx",
    "isDefault": true,
    "managedByTenants": [
      {
        "tenantId": "xxx"
      }
    ],
    "name": "xxx",
    "state": "Enabled",
    "tenantId": "xxx",
    "user": {
      "name": "***",
      "type": "servicePrincipal"
    }
  }
]
/usr/bin/az account set --subscription xxx
/usr/bin/bash /home/vsts/work/_temp/azureclitaskscript1763537136232.sh
WARNING: Command group 'config' is experimental and under development. Reference and support levels: https://aka.ms/CLI_refstatus
Installing Bicep CLI v0.39.26...
Successfully installed Bicep CLI to "/home/vsts/work/_temp/.azclitask/bin/bicep".
Bicep CLI version 0.39.26 (1e90b06e40)

ERROR: WARNING: The following experimental Bicep features have been enabled: Enable extendable parameters. Experimental features should be enabled for testing purposes only, as there are no guarantees about the quality or stability of these features. Do not enable these settings for any production usage, or your production environment may be subject to breaking.
Unhandled exception. System.NotImplementedException: Encountered an unexpected symbol kind '' and type '' when generating a variable access expression.
   at Bicep.Core.Intermediate.ExpressionBuilder.ConvertVariableAccess(VariableAccessSyntax variableAccessSyntax)
   at Bicep.Core.Intermediate.ExpressionBuilder.ConvertWithoutLowering(SyntaxBase syntax)
   at Bicep.Core.Intermediate.ExpressionBuilder.ConvertPropertyAccess(PropertyAccessSyntax propertyAccess)
   at Bicep.Core.Intermediate.ExpressionBuilder.ConvertWithoutLowering(SyntaxBase syntax)
   at System.Linq.Enumerable.SelectArrayIterator`2.MoveNext()
   at System.Collections.Generic.List`1.AddRange(IEnumerable`1)
   at Bicep.Core.Intermediate.ExpressionBuilder.ConvertWithoutLowering(SyntaxBase syntax)
   at Bicep.Core.Intermediate.ExpressionBuilder.ConvertObjectProperty(ObjectPropertySyntax syntax)
   at Bicep.Core.Intermediate.ExpressionBuilder.ConvertObject(ObjectSyntax object)
   at Bicep.Core.Intermediate.ExpressionBuilder.ConvertWithoutLowering(SyntaxBase syntax)
   at Bicep.Core.Intermediate.ExpressionBuilder.Convert(SyntaxBase syntax)
   at Bicep.Core.Emit.ExpressionConverter.ConvertToIntermediateExpression(SyntaxBase syntax)
   at Bicep.Core.Emit.ParameterAssignmentEvaluator.<EvaluateParameter>b__19_0(ParameterAssignmentSymbol parameter)
   at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd(TKey, Func`2)
   at Bicep.Core.Emit.ParameterAssignmentEvaluator.EvaluateParameter(ParameterAssignmentSymbol parameter)
   at Bicep.Core.Emit.EmitLimitationCalculator.CalculateParameterAssignments(SemanticModel model, ParameterAssignmentEvaluator evaluator, IDiagnosticWriter diagnostics)
   at Bicep.Core.Emit.EmitLimitationCalculator.Calculate(SemanticModel model)
   at Bicep.Core.Semantics.SemanticModel.<.ctor>b__20_2()
   at System.Lazy`1.ViaFactory(LazyThreadSafetyMode)
   at System.Lazy`1.ExecutionAndPublication(LazyHelper, Boolean)
   at System.Lazy`1.CreateValue()
   at System.Lazy`1.get_Value()
   at Bicep.Core.Semantics.SemanticModel.get_EmitLimitationInfo()
   at Bicep.Core.Semantics.SemanticModel.GetSemanticDiagnostics()
   at Bicep.Core.Semantics.SemanticModel.AssembleDiagnostics()
   at System.Lazy`1.ViaFactory(LazyThreadSafetyMode)
   at System.Lazy`1.ExecutionAndPublication(LazyHelper, Boolean)
   at System.Lazy`1.CreateValue()
   at System.Lazy`1.get_Value()
   at Bicep.Core.Semantics.SemanticModel.GetAllDiagnostics()
   at Bicep.Core.Semantics.Compilation.GetSourceFileDiagnostics(IOUri sourceFileUri)
   at Bicep.Core.Semantics.Compilation.GetSourceFileDiagnostics(ISourceFile sourceFile)
   at System.Collections.Immutable.ImmutableDictionary.<>c__DisplayClass9_0`3.<ToImmutableDictionary>b__0(TSource element)
   at System.Linq.Enumerable.SelectEnumerableIterator`2.MoveNext()
   at System.Collections.Immutable.ImmutableDictionary`2.AddRange(IEnumerable`1, MutationInput, KeyCollisionBehavior )
   at System.Collections.Immutable.ImmutableDictionary`2.AddRange(IEnumerable`1, Boolean)
   at System.Collections.Immutable.ImmutableDictionary`2.AddRange(IEnumerable`1 )
   at System.Collections.Immutable.ImmutableDictionary.ToImmutableDictionary[TSource,TKey,TValue](IEnumerable`1, Func`2, Func`2, IEqualityComparer`1 , IEqualityComparer`1 )
   at System.Collections.Immutable.ImmutableDictionary.ToImmutableDictionary[TSource,TKey,TValue](IEnumerable`1, Func`2, Func`2)
   at Bicep.Core.Semantics.Compilation.GetAllDiagnosticsByBicepFile()
   at Bicep.Cli.Logging.DiagnosticLogger.LogDiagnostics(DiagnosticOptions options, Compilation compilation)
   at Bicep.Cli.Commands.BuildParamsCommand.Compile(ActiveSourceFileSet workspace, IOUri inputUri, IOUri bicepFileUri, IOUri outputUri, Boolean noRestore, Nullable`1 diagnosticsFormat, Boolean outputToStdOut)
   at Bicep.Cli.Commands.BuildParamsCommand.RunAsync(BuildParamsArguments args)
   at Bicep.Cli.Program.RunAsync(String[] args, CancellationToken cancellationToken)
   at Bicep.Cli.Program.<>c__DisplayClass3_0.<<Main>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at Bicep.Cli.Program.RunWithCancellationAsync(Func`2 runFunc)
   at Bicep.Cli.Program.Main(String[] args)
   at Bicep.Cli.Program.<Main>(String[] args)

##[error]Script failed with exit code: 1

```

--- a/src/Bicep.Cli.IntegrationTests/BuildParamsCommandTests.cs
+++ b/src/Bicep.Cli.IntegrationTests/BuildParamsCommandTests.cs
@@ -1094,6 +1094,8 @@ param objParam object
 
             result.Should().Succeed();
             result.Stdout.Should().NotBeEmpty();
+            var parameters = result.Stdout.FromJson<BuildParamsStdout>().parametersJson.FromJson<JToken>();
+            parameters.Should().HaveValueAtPath("parameters.bar.value", "my-value-foo");
             result.Stderr.Should().Contain("WARNING: The following experimental Bicep features have been enabled: Enable extendable parameters. Experimental features should be enabled for testing purposes only, as there are no guarantees about the quality or stability of these features. Do not enable these settings for any production usage, or your production environment may be subject to breaking.");
             result.ExitCode.Should().Be(0);
         }

